runtime/lua/vim/filetype.lua

```diff
@@ -2489,7 +2489,7 @@ function M.match(args)
 
   -- Finally, check file contents
   if contents or bufnr then
-    contents = contents or M.getlines(bufnr)
+    contents = contents or { unpack(M.getlines(bufnr, 1, 100)), M.getlines(bufnr, -1) }
```

Comment:
This will result in a table of 2 elements since unpack here will only expand to one element

Reply:
Why?

Reply:
That's just how unpack works. It will only expand to multiple if it is placed in the last position.

Read up on lua varargs as a second class citizen if you want to learn more.

Reply:
I've seen this exact pattern on the net; should have known better than to trust the web...

Alternative suggestions welcome.